test/unit/iso_generator_test.rb

```diff
@@ -14,7 +14,20 @@ class IsoGeneratorTest < ActiveSupport::TestCase
       let(:template) { FactoryBot.create(:provisioning_template, template: 'Fake kernel line <%= @kernel %> - <%= @initrd %>') }
 
       setup do
-        host.expects(:provisioning_template).with(kind: :PXELinux).returns(template)
+        host.stubs(:provisioning_template).with(kind: :PXELinux).returns(template)
+      end
+
+      test 'fetch handles redirect' do
+        begin
```

Comment:
You can instead use `Dir.mktmpdir do |dir|` to create a temporary dir that gets automatically deleted at the end of the block.

```ruby
Dir.mktmpdir do |dir|
  # ...
  ForemanBootdisk::ISOGenerator.fetch(File.join(dir, 'bootdisk-test-redirect'), url)
end